proxy/pkg/middleware/account_resolver_test.go

```diff
@@ -0,0 +1,134 @@
+package middleware
+
+import (
+	"context"
+	"github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	"github.com/owncloud/ocis/ocis-pkg/log"
+	"github.com/owncloud/ocis/ocis-pkg/oidc"
+	"github.com/owncloud/ocis/proxy/pkg/config"
+	"github.com/owncloud/ocis/proxy/pkg/user/backend"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestTokenIsAddedWithMailClaim(t *testing.T) {
+	sut := newMockAccountResolver(&userv1beta1.User{
+		Id:   &userv1beta1.UserId{Idp: "https://idx.example.com", OpaqueId: "123"},
+		Mail: "foo@example.com",
+	}, nil)
+
+	req, rw := mockRequest(&oidc.StandardClaims{
+		Iss:   "https://idx.example.com",
+		Email: "foo@example.com",
+	})
+
+	sut.ServeHTTP(rw, req)
+
+	token := req.Header.Get("x-access-token")
```

Comment:
IMO makes sense to use the const from reva for the token header all over the place:
https://github.com/cs3org/reva/blob/314157fa14378d7e8deac6042cf193d11ebbb069/pkg/token/token.go#L29